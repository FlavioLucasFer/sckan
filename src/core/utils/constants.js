const MAIN_COLOR = '#1ac3d6';
const SECUNDARY_COLOR = '#245257';

export {
    MAIN_COLOR,
    SECUNDARY_COLOR,
};