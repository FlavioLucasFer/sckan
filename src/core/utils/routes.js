const HOME_ROUTE = '/';
const LOGIN_ROUTE = '/login';
const COMPANY_ROUTE = '/companies';
const COMPANY_FORM_ROUTE = `${COMPANY_ROUTE}/form`;
const PROJECT_ROUTE = '/projects';
const PROJECT_FORM_ROUTE = `${PROJECT_ROUTE}/form`;
const SPRINT_ROUTE = '/sprints';
const SPRINT_FORM_ROUTE = `${SPRINT_ROUTE}/form`;
const TASK_ROUTE = '/tasks';
const TASK_FORM_ROUTE = `${TASK_ROUTE}/form`;
const USER_ROUTE = '/users';
const USER_FORM_ROUTE = `${USER_ROUTE}/form`;
const USER_PROFILE = '/user';
const USER_SITTINGS = `${USER_PROFILE}/settings`;

export {
	HOME_ROUTE,
  LOGIN_ROUTE,
  COMPANY_ROUTE,
	COMPANY_FORM_ROUTE,
  PROJECT_ROUTE,
	PROJECT_FORM_ROUTE,
  SPRINT_ROUTE,
	SPRINT_FORM_ROUTE,
  TASK_ROUTE,
	TASK_FORM_ROUTE,
  USER_ROUTE,
	USER_FORM_ROUTE,
	USER_PROFILE,
	USER_SITTINGS,
};