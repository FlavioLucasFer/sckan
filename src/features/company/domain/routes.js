'use strict';

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/companies`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	REMOVE: `${BASE_URL}/remove`,
};

export default ROUTES;
