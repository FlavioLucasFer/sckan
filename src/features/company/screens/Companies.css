.new-company-button {
	margin-left: 1.424rem;
}

.list-actions-row {
	/* display: flex; */
	align-items: center;
	/* justify-content: flex-start; */
}

.div-card-container {
	padding: 0 !important;
}

.card-container {
	min-height: 70vh;
}

.data-card {
	position: relative;
	width: 100%;
	min-height: 300px;
}

.no-selected-company-message {
	text-align: center;
	margin: auto;
}

.actions {
	position: absolute;
	right: 12px;
	top: 0px;
}

.actions>i {
	padding: 10px;
	text-align: center;
	cursor: pointer;
}

.actions>i:hover {
	background-color: rgba(0, 0, 0, .1);
}

.company-informations-container>h5 {
	margin-top: 0;
}

.company-informations-container>p {
	font-size: 18px;
}

.company-informations-container>p>span {
	font-weight: bold;
}
