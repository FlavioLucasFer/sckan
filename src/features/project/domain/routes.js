'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/projects`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
};

export default ROUTES;
