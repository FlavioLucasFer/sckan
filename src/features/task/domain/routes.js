'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/tasks`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
};

export default ROUTES;
