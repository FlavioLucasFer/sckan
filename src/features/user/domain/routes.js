'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/users`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	SIGN_IN: `${BASE_URL}/sign-in`,
	LOGOUT: `${BASE_URL}/logout`,
};

export default ROUTES;
