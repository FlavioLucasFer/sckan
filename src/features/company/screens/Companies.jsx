import './Companies.css';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import ListScreen from 'core/components/ListScreen';
import { COMPANY_FORM_ROUTE } from 'core/utils/routes';

import companyService from 'features/company/domain/service';

import { setCompanies } from 'redux/reducers/companies';

function Companies() {
	const dispatch = useDispatch();
	const companies = useSelector(state => state.companies.value);

	const fields = [
		{ title: 'Company name', field: 'companyName' },
		{ title: 'Trading name', field: 'tradingName' },
		{ title: 'Federal document', field: 'federalDocument' },
		{ title: 'E-mail', field: 'email' },
	];

	useEffect(async () => {
		await getCompanies();
	}, []);

	async function getCompanies() {
		try {
			const companies = await companyService.find();
			dispatch(setCompanies(companies));
		} catch (err) {
			console.log('err:', err);
		}
	}

	async function handleRemove(id) {
		try {
			await companyService.remove(id);
			await getCompanies();
		} catch (err) {
			console.log('err:', err);
		}
	}

	return (
		<ListScreen title="Companies"
			singularTitle="Company"
			data={companies}
			fields={fields}
			identifierField="companyName"
			itemsQuantityByPage={10}
			formRoute={COMPANY_FORM_ROUTE}
			onRefresh={getCompanies}
			onRemove={handleRemove} />
	);
}

export default Companies;
