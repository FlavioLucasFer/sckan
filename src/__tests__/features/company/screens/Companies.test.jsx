import React from 'react';
import { shallow } from 'enzyme';

import Companies from 'features/company/screens/Companies';
import { COMPANY_FORM_ROUTE } from 'core/utils/routes';

const mockHistory = {
	push: jest.fn(),
};

jest.mock('react-router-dom', () => ({
	...(jest.requireActual('react-router-dom')),
	useHistory: () => (mockHistory),
}));

describe('Testing Companies screen component', () => {
	afterAll(() => {
		jest.clearAllMocks();
	});
	
	it('should render correctly', () => {
		const wrapper = shallow(
			<Companies />
		);

		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('should navigate to company form screen when clink on "add" button', () => {
		const wrapper = shallow(
			<Companies />
		);

		wrapper.find({ formRoute: COMPANY_FORM_ROUTE })
			.dive()
			.find('.new-item-button')
			.simulate('click')

		expect(mockHistory.push).toBeCalledWith(COMPANY_FORM_ROUTE);
	});
});
