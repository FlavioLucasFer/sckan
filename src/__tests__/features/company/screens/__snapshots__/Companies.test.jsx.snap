// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Companies screen component should render correctly 1`] = `"<ListScreen title=\\"Companies\\" singularTitle=\\"Company\\" data={{...}} fields={{...}} identifierField=\\"companyName\\" pagesQuantity={3} formRoute=\\"/companies/form\\" onRefresh={{...}} />"`;
